hippunfold/workflow/rules/atlas_gen.smk

```diff
@@ -0,0 +1,332 @@
+            root=work,
+            datatype="warps",
+            suffix="invwarp.nii.gz",
+            from_="unfold",
+            to="avgunfold",
+            hemi="{hemi}",
+            label="{label}",
+            **inputs.subj_wildcards,
+        ),
+    group:
+        "subj"
+    shell:
+        "cp {params.glob_input_warp} {output.warp} && "
+        "cp {params.glob_input_invwarp} {output.invwarp}"
+
+
+rule copy_avgtemplate_metric:
+    """ copy avgtemplate metric, adjusting header to match the original data
+     (since this seems to get garbled in z by ants)"""
+    input:
+        avgtemplate_dir="template/avgtemplate_{label}",
+        ref_metric=lambda wildcards: inputs[config["modality"]].expand(
+            bids(
+                root=work,
+                datatype="anat",
+                suffix="{metric}.nii.gz",
+                space="unfold2d",
+                hemi="{hemi}",
+                label="{label}",
+                **inputs.subj_wildcards,
+            ),
+            label=wildcards.label,
+            metric=wildcards.metric,
+            **expand_hemi(),
+        )[0],
+    params:
+        in_metric=lambda wildcards, input: "{avgtemplate_dir}/template{i}.nii.gz".format(
+            avgtemplate_dir=input.avgtemplate_dir,
+            i=reg_metrics.index(wildcards.metric),
+        ),
+    output:
+        metric="template/avgtemplate_label-{label}_{metric}.nii.gz",
+    script:
+        "../scripts/set_metric_nii_header.py"
+
+
+def get_metric_template(wildcards, input):
+
+    num_modalities = len(reg_metrics)
+
+    warp_prefix = "template/warp_label-{label}_"
+    metrics = sorted(glob(f"{warp_prefix}_template?.nii.gz"))
+    return metrics
+
+
+### final atlas files ###
+
+
+ruleorder: gen_atlas_surfs > download_extract_atlas
+ruleorder: avg_metrics > download_extract_atlas
+ruleorder: maxprob_subfields > download_extract_atlas
+
+
+rule gen_atlas_surfs:
+    input:
+        metric_nii=expand(
+            "template/avgtemplate_label-{label}_{metric}.nii.gz",
+            metric=reg_metrics,
+            allow_missing=True,
+        ),
+    params:
+        z_level=get_unfold_z_level,
+    output:
+        midthickness_surf=directory(Path(download_dir) / "atlas" / "{atlas}")
+        / config["atlas_files"]["{atlas}"]["surf_gii"],
+    container:
+        config["singularity"]["autotop"]
+    conda:
+        conda_env("pyvista")
+    script:
+        "../scripts/surf_gen.py"
+
+
+rule gen_atlas_densities:
+    input:
+        midthickness_surf=directory(Path(download_dir) / "atlas" / "{atlas}")
+        / config["atlas_files"]["{atlas}"]["surf_gii"],
+    output:
+        ref_unfold=os.path.join(
+            workflow.basedir,
+            "..",
+            "resources",
+            "unfold_template_{label}",
+            "tpl-avg_space-unfold_den-{density}_midthickness.surf.gii",
+        ),
+    shell:
+        "cp {input} {output}"  # TODO get actual densities
+
+
+rule avg_metrics:
```

Comment:
Not sure what you're trying to average over here, as the template#.nii.gz files are already averages, each being a different channel  -- anyways I fixed this in my previous commits will revert back